migrator/zenodo_rdm_migrator/actions/transform/oauth.py

```diff
@@ -222,5 +225,189 @@ def _transform_data(self):
+    """Zenodo to RDM OAuth client linked account connect action."""
+
+    name = "oauth-application-connect"
+    load_cls = load.OAuthLinkedAccountConnectAction
+
+    @classmethod
+    def matches_action(cls, tx):
+        """Checks if the data corresponds with that required by the action."""
+        if 3 > len(tx.operations) or len(tx.operations) > 7:
+            return False
+
+        mandatory = {
+            "oauthclient_remoteaccount",
+            "oauthclient_remotetoken",
+            "oauthclient_useridentity",
+        }
+
+        optional = {
+            "oauth2server_client",
+            "oauth2server_token",
+        }
+
+        allow_updates = {"oauthclient_remoteaccount"}
+
+        for op in tx.operations:
+            if op["op"] == OperationType.UPDATE:
+                # nested to be able to have the final else clause by op type
+                if op["source"]["table"] not in allow_updates:
+                    return False
+            elif op["op"] == OperationType.INSERT:
+                try:
+                    mandatory.remove(op["source"]["table"])
+                except KeyError:
+                    try:
+                        optional.remove(op["source"]["table"])
+                    except KeyError:
+                        return False
+            else:
+                return False
+
+        # all mandatory were found and optionals are all or none
+        return len(mandatory) == 0 and (len(optional) == 2 or len(optional) == 0)
+
+    def _transform_data(self):
+        """Transforms the data and returns dictionary."""
+        remote_account = None
+        remote_token = None
+        server_client = None
+        server_token = None
+        user_identity = None
+
+        for op in self.tx.operations:
+            if op["source"]["table"] == "oauthclient_remoteaccount":
+                if op["op"] == OperationType.INSERT:
+                    remote_account = op["after"]
+                else:
+                    for key, value in op["after"].items():
+                        remote_account[key] = value
+            elif op["source"]["table"] == "oauthclient_remotetoken":
+                remote_token = op["after"]
+            elif op["source"]["table"] == "oauthclient_useridentity":
+                user_identity = op["after"]
+            elif op["source"]["table"] == "oauth2server_client":
+                server_client = op["after"]
+            elif op["source"]["table"] == "oauth2server_token":
+                server_token = op["after"]
+
+        result = {
+            "tx_id": self.tx.id,
+            "remote_account": IdentityTransform()._transform(remote_account),
+            "remote_token": IdentityTransform()._transform(remote_token),
+            "user_identity": IdentityTransform()._transform(user_identity),
+        }
+        if server_client:
+            result["server_client"]: OAuthServerTokenTransform()._transform(
+                server_token
```

Comment:
typo: 
```suggestion
                server_client
```

change the following line to the opposite. How did it pass tests!???